### PR TITLE
Fix #12 wiki image and wiki links

### LINKED ISSUE #12: The front-matter is not parsed if the tags are in YAML
## Issue

This is the follow-up issue of #5 

The front-matter is as like below. The `tags` is in YAML format, and it's in between of other metadata. 

```
---
title: No More Basic Auth &#58; htpasswd with OAuth2 Proxy
date: 2024-12-15 20:44:14
author: arul
category: Security
tags:
  - proxy
  - Traefik
  - APISix
  - nginx
slug: no-more-basic-auth-htpasswd-with-oauth2-proxy
cover: /assets/images/basic-auth-oauth2-proxy-cover.png
color: gray
headline: Create cookie session store for htpasswd based authentication using oauth2-proxy. This will eliminate the insecure basic authentication.
status: published
---
```

Whatever metadata after `tags` is not parsing. This article automatically moved to drafts, since the published metadata is at the end. 

## Workaround

Now I have changed the order and placed the `tags` at the end. 

```
---
title: No More Basic Auth &#58; htpasswd with OAuth2 Proxy
date: 2024-12-15 20:44:14
author: arul
category: Security
slug: no-more-basic-auth-htpasswd-with-oauth2-proxy
cover: /assets/images/basic-auth-oauth2-proxy-cover.png
color: gray
headline: Create cookie session store for htpasswd based authentication using oauth2-proxy. This will eliminate the insecure basic authentication.
status: published
tags:
  - proxy
  - Traefik
  - APISix
  - nginx
---
```
Now it's working as expected. 

## What Next?

Pelican talks about plugin [ordering](https://docs.getpelican.com/en/latest/plugins.html#how-to-create-plugins) in `PLUGINS`. But it's not working. 

Maybe the metadata received to this plugin without others. 

FYI, The unit test is passing. Because it takes care about only `ObsidianMarkdownReader`

### PR DESCRIPTION
**Pull request by arulrajnet**
This has the fix for #12 

- added support for windows
- image support for webp, avif, jpeg
- Added logging and enabled

Unit test passed

image

After the fix image and links are working.

### CODE REVIEW

**Comment by arulrajnet** [comment]
@jonathan-s Could you review and merge this?

File: pelican/plugins/obsidian/obsidian.py [hunk 1]
```diff
@@ -98,10 +108,19 @@ def read(self, source_path):
         Parse content and metadata of markdown files.
         Also changes the links to the acceptable format for pelican
         """
+        with pelican_open(source_path) as text:
+            m = HEADER_RE.fullmatch(text)
+
+        if not m:
+            return super().read(source_path)
+
+        text = m.group("content")
```
Comment (jonathan-s) [inline pelican/plugins/obsidian/obsidian.py:117]: @arulrajnet Any chance you could explain why we need to do more or less the same thing here as we do in `YamlMetadataReader` and we can't use the contents of super directly? In other words, what is the edge case we solve for here? 

If there is a particular edge case here, can we add it to tests?

File: pelican/plugins/obsidian/obsidian.py [hunk 2]
```diff
@@ -125,20 +144,33 @@ def populate_files_and_articles(article_generator):
...
+        # For windows
+        if os.sep == '\\':
+            path = path.replace('\\', '/')
 
         # This work on both pages and posts/articles
         ARTICLE_PATHS[filename] = path
 
+    __log__.debug('Found %d articles', len(ARTICLE_PATHS))
+
     # Get list of all other relavant files
```
Comment (jonathan-s) [inline pelican/plugins/obsidian/obsidian.py:156]: Not your typo, but would be nice to fix :)

File: pelican/plugins/obsidian/obsidian.py (hunk 1 above)
Comment (arulrajnet) [reply pelican/plugins/obsidian/obsidian.py:117]: Check this comment https://github.com/jonathan-s/pelican-obsidian/issues/12#issuecomment-2563904562

## The issue

The wiki links and wiki images are not working after adding `YamlMetadataReader` which is the main use case of this plugin.

For example

Markdown file, have the following content

```
![[pelican-in-rock.png]]
```
The expected output is, the image has to be found and loaded as HTML. 

But the actual output is ``. It's loaded as `md` not the HTML. 

## Root cause

The `super().read(source_path)` is converted the markdown to HTML. Then we replaced the wiki/obsidian links to markdown links in `self.replace_obsidian_links(content)`. 

That's why, we read the content as markdown `m.group("content")` as like `YamlMetadataReader` . Then replace the links. Then convert to HTML. 

The core future of this plugin is broken, and we fixed in this PR. 

## Other fixes

* This plugin is not works in Windows machine. Fixed that
* Added support for modern image formats `webp, avif` 
* Enabled logging.

File: pelican/plugins/obsidian/obsidian.py (hunk 2 above)
Comment (arulrajnet) [reply pelican/plugins/obsidian/obsidian.py]: done.

File: pelican/plugins/obsidian/obsidian.py (hunk 1 above)
Comment (arulrajnet) [reply pelican/plugins/obsidian/obsidian.py:117]: Added unit test for the content generation.  The following cases added

- Internal link
- Internal image
- Internal link but it doesn't exists

image

**Comment by jonathan-s** [comment]
Thanks for adding the testcases, and thanks for taking the time to fix it!
